File: FashionMNIST/Telum/classifier.py
from PyRuntime import OMExecutionSession
import json
import numpy as np
model = "fashion_MNIST/fashion_classifier.so"

convert_types = {"f32":"float32",
                 "f16":"float16"}

def inference(image):
    session = OMExecutionSession(model)
    input_signature_json = json.loads(session.input_signature())
    signature = input_signature_json[0]
    input_type = signature["type"]

    image = image[np.newaxis,np.newaxis,...].astype(convert_types[input_type])
    imageset = []
    imageset.append(image)
    output = {}
    output["output"] = session.run(imageset)

    return output

def main():
    import sys
    from imageio import readpgm
    from termshow import show, ANSI_COLOURS

    iname = "test.pgm"

    classes = ("T-shirt", "Trouser", "Pullover", "Dress", "Coat", "Sandal", "Shirt", "Sneaker", "Bag", "Ankle boot",)

    if len(sys.argv) > 1:
        iname = sys.argv[1]

    c, test_image = readpgm(iname)

    print(f"Loaded image: {iname}")
    show(test_image, ANSI_COLOURS)

    results = inference(test_image)
 
    print(f"Expected: {c}\nPredicted: {classes[np.argmax(results['output'])]}")

if __name__ == "__main__":
    main()

File: FashionMNIST/Telum/classifier.java
import com.ibm.onnxmlir.OMModel;
import com.ibm.onnxmlir.OMTensor;
import com.ibm.onnxmlir.OMTensorList;

import imageio.pgm;

public class classifier {

    public static float[] inference(pgm image) {
        float[] ret_val = {0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
        return ret_val;
    }

    public static int max_at(float[] data) {
        int max_loc = 0;

        for (int i = 0; i < data.length; i++) {
            if (data[i] > max_loc) {
                max_loc = i;
            }
        }

        return max_loc;
    }

    public static void main(String[] args) {
        String[] fashion_classes = {"T-shirt", "Trouser", "Pullover", "Dress", "Coat", "Sandal", "Shirt", "Sneaker", "Bag", "Ankle boot"};

        if (args.length == 0) {
            System.out.println("java classifier <image file>");
            System.exit(0);
        }

        pgm image = new pgm();
        image.read_image(args[0]);

        System.out.println("Loaded Image: " + args[0]);
        image.show_image();
        float[] results = inference(image);

        String prediction = fashion_classes[max_at(results)];

        System.out.println("Expected: " + image.get_classification());
        System.out.println("Predicted: " + prediction);

    }
}